graphql/parser.go

```diff
@@ -119,43 +136,59 @@ func parseSelectionSet(input *ast.SelectionSet, globalFragments map[string]*Frag
 				return nil, err
 			}
 
-			selections = append(selections, &Selection{
+			newSelection := &Selection{
 				Alias:        alias,
 				Name:         selection.Name.Value,
 				UnparsedArgs: args,
 				SelectionSet: selectionSet,
-			})
+			}
+
+			if len(selection.Directives) > 0 {
+				newSelection.Directives = directives
```

Comment:
does this need to be conditional? if it's empty, it seems okay to be nil or empty slice?

Reply:
Currently, it's expected to be nil, but without the conditional, it will set it to an empty slice, which slightly modifies our expected output. This breaks a lot of our tests, and since the length of the list of directives is zero anyway, I figured it was safe to not set that value.